### DISCUSSION TITLE
Questions and/or feature requests as a test user/developer being pleasantly surprised!

### DISCUSSION THREAD

**Discussion by bencresty** [plausible/analytics#924] [Anything Else]
Hi, 

First of all I am very pleasantly surprised about Plausible. Both the interface as the documentation and usage are very refreshing and well made! Thanks for being a nice alternative to the big bad wolf 🥇 

Also like it that everything 'just' works! Even when running from local development environments (unlike google analytics, where that isn't that straight forward). Everything is so waaaaaay more nice and makes me feel great when using it compared to Googles! Great work. At first it felt a little too stripped down, but quickly I found how powerful it really is and how much I really was missing in Google Analytics in all the noise there!

After digging through the main things, used it as a developer now and coming from years of using Google Analytics, I have some questions cq feature requests though. As I am curious as a test driver to know what you think of these things and/or if they will be supported in the (near) future. Sorry if some of the points are already asked.

I would be very much interested to know if the following are possible cq will be possible in the future, or what your thoughts are about these (espectially asking the creators of the Plausible, but open to every answer/help!):

**most important / really missing now / questions**
- See a route an anonymous user made from page to page? This is what I'm currently missing the most; it tells a lot about where a website or app needs improvement if we can see how visitors 'walk through' the website.
- The same with seeing how long a user stays on the website (clicking through it). When a user only visits one blog article and therefor has a bounce rate of 100%; it's still a very great visitor if he/she was reading that blog or tutorial for an hour! How do we know how long a visitor is on a page? And how do we know how long he was on the website during a session? Is that possible to see? Especially in relation the which pages she/he visited.
- Is it possible to see what place a visitor is from? Seeing visitors' country is great, but when doing local business it would be very nice if we could see cities/places too. Visitors from the same country, but far way, are most likely not gonna become customers.
- I added a link to Google Search Console, but I'm not sure about Google Ads. I see 'Campaign' in the Plausible interface, but don't see any data there (yet). Is it somehow possible to (get to) know if visitors came from clicking on an Ad? That's pretty important here to know if ads work or need adjustments.
- When having several properties on a custom event I didn't manage to find a way in the interaface to see all these properties of a single event at once. So we are missing the connection between the properties of a single event, like what locale a user was in on a special event, where one of the properties was a or b? AFAIK we can only see one property at the time and are missing the relation between the properties, so missing important information. I'd rather not combine properties into one as it's also nice to be able to see the properties seperately. Am I missing something here or is it impossible to see properties at once for a single event? Or query on properties as an alternative?
- When using Google Analytics we needed to send some 'pulses', like on the scroll events, to 'keep the connection alive', so it doesn't start new sessions. How is that with Plausible? Do we need to send some trigger once in a while while a visitor is on the page to keep it counted as a single session? And is there some concept of sessions to begin with?

**nice to have/questions**
- Is it possible to see analytics of overal new visitors vs returning visitors?
- Is it possible to zoom in on the map? (if not; it would be great if we could use a mouse wheel to zoom in/out to make click areas bigger on especially the little countries)
- When adding custom events to site A I often like to use them also on site B. Like a staging website and a production website sharing the same events, but also websites that are alike. It would be nice if there's a way to copy and paste events from one to another domain. Or perhaps even better; write to and read them from file. Is there something like that present?
- It would be nice if we could give pageview events names too. Like 'Visited the contact page'.
- I understand we can use wildcards in pageviews, which is great. But I was missing a way to add a single page-view event to track visitors that were on the `/contact` page OR the `/en/contact` page. I couldn't find a way to do this with one pageevent. Something like being able to use a regex for this would be a great as now the data on several pages are all duplicated and hard to filter. Or am I missing something here?
- When having different domains that are connected to each other, like a `dev` - `staging` and `production` site, or sites that belong to the same type of website, it would be nice if we could group websites in the websites overview page. With every website having 3 environment stages and could have multiple sub sites next to this with the same environments, it quickly becomes crowded I think.
- When adding goals I was surprised to see the list is no on alfabetical order, making it difficult to find an already created goal in a long list. Is there a way to sort the list? 
- I couldn't find anything in the docs on the path to the script that also tracks outbound-links when loading ths script via the custom (sub)domain. I succesfully guessed that `index.js` could be replaced by `plausible.outbound-links.js` too when loading it from own subdomain and it works fine. Is this the right way? I'm asking this as I was hoping for a more 'under the radar' script name like the default `index.js`, something like 'index-plus.js' or whatever. That, to prevent that browser blockers (in the future) recognise the `plausible` in the filename and decide to block the script. That chance might be very small when having a more general name like `index.js` that will probably never get blocked, because else they should do file sniffing to understand it's from plausible.

**pricing / while developing**
- I think the pricing system pretty decent. However, I think (and it's holding me back a little to jump in) it would be welcome if development visits wouldn't count for the amount of page views used to calculate the price. I like the way the Craft CMS is doing this; it's a payed CMS, but you only have to pay for a live website. If a website is on an obvious development or staging domain, you don't have to pay. So Craft does that by filtering on subdomains like 'dev', 'staging', 'development' or domains like 'localhost' etc. I think it would be fair that when developing a website you don't (yet) have to pay for the analytics as it's only for creating a product for a customer that uses and pays for the analytics after that. That way developers don't necesarily need to pay monthly for analytics when not every customer wants analytics. What are your thoughts on this?
- Is it possible to pause an account/payment? Not all customers want analytics so sometimes we don't work with analytics for months. Do we still need to pay all these months as developers? See previous point; I think it would be best if developers shouldn't pay while developing a website for a client (or doing research/upgrade knowledge) and once a client starts using it the client pays the monthly fees.
- Please understand when being a one man company these things count. Especially during these times now unfortunately.

Sorry for the long post, but it's out of enthousiasm and curiosity, but also because I like to become a client and/or provider for clients, so some of these questions/requests are important stuff to know to make a choice.

Thanks in advance!

**Comment by metmarkosaric** [comment]
thanks @bencresty for the kind words! most of these have their own threads already as that's better way to discuss feature requests. i recommend you use the search functionality to explore the different threads. have included some links below

>     * See a route an anonymous user made from page to page? This is what I'm currently missing the most; it tells a lot about where a website or app needs improvement if we can see how visitors 'walk through' the website.

User flows #149 

>     * The same with seeing how long a user stays on the website (clicking through it). When a user only visits one blog article and therefor has a bounce rate of 100%; it's still a very great visitor if he/she was reading that blog or tutorial for an hour! How do we know how long a visitor is on a page? And how do we know how long he was on the website during a session? Is that possible to see? Especially in relation the which pages she/he visited.

Time on page #111 

>     * Is it possible to see what place a visitor is from? Seeing visitors' country is great, but when doing local business it would be very nice if we could see cities/places too. Visitors from the same country, but far way, are most likely not gonna become customers.

Cities #102 

>     * I added a link to Google Search Console, but I'm not sure about Google Ads. I see 'Campaign' in the Plausible interface, but don't see any data there (yet). Is it somehow possible to (get to) know if visitors came from clicking on an Ad? That's pretty important here to know if ads work or need adjustments.

You can read more about how to use UTM tags here https://plausible.io/docs/manual-link-tagging

>     * When having several properties on a custom event I didn't manage to find a way in the interaface to see all these properties of a single event at once. So we are missing the connection between the properties of a single event, like what locale a user was in on a special event, where one of the properties was a or b? AFAIK we can only see one property at the time and are missing the relation between the properties, so missing important information. I'd rather not combine properties into one as it's also nice to be able to see the properties seperately. Am I missing something here or is it impossible to see properties at once for a single event? Or query on properties as an alternative?

That's not possible at this time #846 

>     * When using Google Analytics we needed to send some 'pulses', like on the scroll events, to 'keep the connection alive', so it doesn't start new sessions. How is that with Plausible? Do we need to send some trigger once in a while while a visitor is on the page to keep it counted as a single session? And is there some concept of sessions to begin with?

Plausible is based on unique visitors and pageviews. No matter how many times the same person visits the site on the same day, we still show 1 unique visitor and have no concept of sessions in the UI.

>     * Is it possible to see analytics of overal new visitors vs returning visitors?

No, we are a privacy first platform so we don't use cookies nor any other persistent identifiers. See https://plausible.io/data-policy 

>     * Is it possible to zoom in on the map? (if not; it would be great if we could use a mouse wheel to zoom in/out to make click areas bigger on especially the little countries)

That's not possible #168 

>     * When adding custom events to site A I often like to use them also on site B. Like a staging website and a production website sharing the same events, but also websites that are alike. It would be nice if there's a way to copy and paste events from one to another domain. Or perhaps even better; write to and read them from file. Is there something like that present?

At this time you need to manually create different goals on all of your sites.

>     * It would be nice if we could give pageview events names too. Like 'Visited the contact page'.

Pageview names are based on the names of individual pages so that's pretty much how it ends up being as long as your contact page is called contact.

>     * I understand we can use wildcards in pageviews, which is great. But I was missing a way to add a single page-view event to track visitors that were on the `/contact` page OR the `/en/contact` page. I couldn't find a way to do this with one pageevent. Something like being able to use a regex for this would be a great as now the data on several pages are all duplicated and hard to filter. Or am I missing something here?

Custom filter #774 

>     * When having different domains that are connected to each other, like a `dev` - `staging` and `production` site, or sites that belong to the same type of website, it would be nice if we could group websites in the websites overview page. With every website having 3 environment stages and could have multiple sub sites next to this with the same environments, it quickly becomes crowded I think.

Grouping of sites #837 

>     * When adding goals I was surprised to see the list is no on alfabetical order, making it difficult to find an already created goal in a long list. Is there a way to sort the list?

There's no way to sort it at this time. #458 

>     * I couldn't find anything in the docs on the path to the script that also tracks outbound-links when loading ths script via the custom (sub)domain. I succesfully guessed that `index.js` could be replaced by `plausible.outbound-links.js` too when loading it from own subdomain and it works fine. Is this the right way? I'm asking this as I was hoping for a more 'under the radar' script name like the default `index.js`, something like 'index-plus.js' or whatever. That, to prevent that browser blockers (in the future) recognise the `plausible` in the filename and decide to block the script. That chance might be very small when having a more general name like `index.js` that will probably never get blocked, because else they should do file sniffing to understand it's from plausible.

You can read more on how to use external link click tracking on a custom domain here https://plausible.io/docs/outbound-link-click-tracking#if-youre-serving-our-script-from-your-custom-domain

>     * I think the pricing system pretty decent. However, I think (and it's holding me back a little to jump in) it would be welcome if development visits wouldn't count for the amount of page views used to calculate the price. I like the way the Craft CMS is doing this; it's a payed CMS, but you only have to pay for a live website. If a website is on an obvious development or staging domain, you don't have to pay. So Craft does that by filtering on subdomains like 'dev', 'staging', 'development' or domains like 'localhost' etc. I think it would be fair that when developing a website you don't (yet) have to pay for the analytics as it's only for creating a product for a customer that uses and pays for the analytics after that. That way developers don't necesarily need to pay monthly for analytics when not every customer wants analytics. What are your thoughts on this?

For us the cost is the same of a pageview of a live visitor or pageview of you in staging. You can always exclude tracking from non-live pages or reset the stats before you make the site live in order to not have those count towards billable pageviews.

>     * Is it possible to pause an account/payment? Not all customers want analytics so sometimes we don't work with analytics for months. Do we still need to pay all these months as developers? See previous point; I think it would be best if developers shouldn't pay while developing a website for a client (or doing research/upgrade knowledge) and once a client starts using it the client pays the monthly fees.

For us the cost is just as much in the storage of old analytics as in counting new analytics so this would not be feasible. But you can always cancel your subscription and remove tracking scripts when you don't want to use Plausible. You can then always restart your subscription when you want to use analytics again. It's easy to upgrade/downgrade in our self-serve platform

**Reply by bencresty** [reply]
Hi @metmarkosaric thanks for the quick response and clear answers/references. Sorry for not having found these threads before. Very understandable standpoint on the price also.

> But you can always cancel your subscription and remove tracking scripts when you don't want to use Plausible. You can then always restart your subscription when you want to use analytics again. It's easy to upgrade/downgrade in our self-serve platform

That's a good point, the analytics of my own companies' website will still be needed, even if there are no customers wanting to use Plausible for a year I need to develop and test for, so than I'd still need a subscription plan. So the only question left on this point is how quickly can we upgrade/downgrade/cancel subscription plans.

I found these docs:
https://plausible.io/docs/change-plan
https://plausible.io/docs/cancel-subscription

But I couldn't find information about the notice period or consequences of changing or cancelling a plan.
- Can we change/cancel plans last day right before renewal date? 
- And is it as simple as monthly based is per the first of the month and yearly is per the first of the year?
- Can we do unlimited upgrades and downgrades per the subscription durations?
- Is it possible to cancel or upgrade/downgrade a subscription plan half way the plans duration?

Sorry for all these questions, but not all my customers want analytics because of the type of products I create isn't conventional websites, but when they do I really would like to work with a product like Plausible rather than Google and be an advocate on it! Thanks again

**Reply by metmarkosaric** [reply]
Sure, no problem! 

> * Can we change/cancel plans last day right before renewal date?

Yes, you can change/cancel plans at any time including on the day before the renewal date. You don't need to contact us at any stage. It is all quick to do and self-serve.

> * And is it as simple as monthly based is per the first of the month and yearly is per the first of the year?

It is based on the date your subscribed in the first place so not necessary on the first of the month. If you subscribe on say today April 13th then your subscription runs for one month/year from today.
 
> * Can we do unlimited upgrades and downgrades per the subscription durations?

You can upgrade or downgrade at any time. There are no limits.

> * Is it possible to cancel or upgrade/downgrade a subscription plan half way the plans duration?

Yes. Our payment processor will do the prorated calculation no matter when you upgrade/downgrade/cancel to reflect the amount you have already paid and the time until your current subscription is supposed to expire.

**Reply by bencresty** [reply]
Thanks again, sounds very flexible. You guys are great!

**Reply by metmarkosaric** [reply]
thank you! enjoy Plausible!
